spec/requests/tags_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+RSpec.describe "Tags" do
+  it_behaves_like "requests that assert edition state",
+                  "tagging a non editable edition",
+                  routes: { tags_path: %i[get patch] } do
+    let(:edition) { create(:edition, :published) }
+  end
+
+  describe "GET /documents/:document/tags" do
+    let(:tag_field) { build(:tag_field, type: "multi_tag") }
+    let(:edition) do
+      document_type = build(:document_type, tags: [tag_field])
+      create(:edition, document_type_id: document_type.id)
+    end
+
+    it "returns successfully" do
+      tag_name = SecureRandom.hex(8)
+      stub_publishing_api_has_linkables(
+        [{ "content_id" => SecureRandom.uuid, "internal_name" => tag_name }],
+        document_type: tag_field.document_type,
+      )
+      get tags_path(edition.document)
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to have_content(tag_name)
+    end
+
+    it "returns service unavailable when the Publishing API is down" do
+      stub_publishing_api_isnt_available
+
+      get tags_path(edition.document)
+
+      expect(response).to have_http_status(:service_unavailable)
+      expect(response.body).to have_content(I18n.t!("tags.edit.api_down"))
+    end
+
+    it "shows a warning when editing the primary organisation tag of an access limited edition" do
+      tag_field = build(:tag_field,
+                        type: "single_tag",
+                        id: "primary_publishing_organisation",
+                        document_type: "organisation")
+      document_type = build(:document_type, tags: [tag_field])
+      edition = create(:edition,
+                       :access_limited,
+                       document_type_id: document_type.id,
+                       created_by: current_user)
+      stub_publishing_api_has_linkables(
+        [{ "content_id" => current_user.organisation_content_id,
+           "internal_name" => "Organisation" }],
+        document_type: tag_field.document_type,
+      )
+
+      get tags_path(edition.document)
+      expect(response).to have_http_status(:ok)
+      expect(response.body)
+        .to have_content(I18n.t!("tags.edit.organisation_warning"))
+    end
+  end
+
+  describe "PATCH /documents/:document/tags" do
+    before { stub_any_publishing_api_put_content }
+
+    it "redirects to document path on successful request" do
+      tag_field = build(:tag_field, type: "multi_tag", id: "field")
```

Comment:
Minor: do we need any tag fields for this to work?

Reply:
No, it doesn't I've removed this. Thanks